core/src/view/flyTo.cpp

```diff
@@ -0,0 +1,85 @@
+#include "view/flyTo.h"
+
+#include "util/mapProjection.h"
+#include "view/view.h"
+#include <cmath>
+#include "log.h"
+
+namespace Tangram {
+
+float getMinimumEnclosingZoom(double aLng, double aLat, double bLng, double bLat, const View& view, float buffer) {
+    const MapProjection& projection = view.getMapProjection();
+    glm::dvec2 aMeters = projection.LonLatToMeters(glm::dvec2(aLng, aLat));
+    glm::dvec2 bMeters = projection.LonLatToMeters(glm::dvec2(bLng, bLat));
+    double distance = glm::distance(aMeters, bMeters) * (1. + buffer);
+    double focusScale = distance / (2. * MapProjection::HALF_CIRCUMFERENCE);
+    double viewScale = view.getWidth() / projection.TileSize();
+    double zoom = -log2(focusScale / viewScale);
+    return zoom;
+}
+
+std::function<glm::dvec3(float)> getFlyToFunction(const View& view, glm::dvec3 start, glm::dvec3 end, double& _duration) {
+
+    // User preference for zoom/move curve sqrt(2)
+    const double rho = 1.414;
+
+    const double scale = std::pow(2.0, end.z - start.z);
+
+    // Current view bounds in Mercator Meters
+    auto rect = view.getBoundsRect();
+    auto width = std::abs(rect[0][0] - rect[1][0]);
+    auto height = std::abs(rect[0][1] - rect[1][1]);
+
+    const double w0 = std::max(width, height);
+    const double w1 = w0 / scale;
+
+    const glm::dvec2 c0{start.x, start.y};
+    const glm::dvec2 c1{end.x, end.y};
+
+    const double u1 = glm::distance(c0, c1);
+
+    auto b = [=](int i) {
```

Comment:
Will be good to document what these `b`, `r`, `u`, `w` and `S` represent here. One will save time to not refer to equation 9 on the implemented paper.